### ISSUE TITLE
Style Construction

### ISSUE THREAD

**Issue by KingOfBrian** [Raizlabs/BonMot#179]
Constructing Styles in BonMot should be easy and have good tooling support with Xcode. There are 3 methods of creating styles listed below.

# Defaulted Initializer
This implementation provides a giant initializer with everything defaulted to `nil` or `[]`. This reads nicely, but the Xcode autocompletion is disappointing. 

```swift
let style = BonMotI(
    font: .fontB,
    textColor: .colorB,
    headIndent: 10,
    tailIndent: 10
)
```
<img width="1155" alt="bonmot-defaultedinitializer" src="https://cloud.githubusercontent.com/assets/114322/18294256/e83d255c-7465-11e6-99ea-f25ad75cb961.png">

# Block configuration (Builder pattern)
This implementation provides an `inout` closure to configure the style like a normal struct. This reads pretty nicely, and feels Swifty, but it is a bit wordy.

```swift
let style = BonMotC { style in
    style.font = .fontB
    style.textColor = .colorB
    style.headIndent = 10
    style.tailIndent = 10
}
```
<img width="858" alt="bonmot-builder" src="https://cloud.githubusercontent.com/assets/114322/18294240/d6fb46b6-7465-11e6-86ec-383b3864305d.png">

# Part construction
This implementation breaks up every property of the `AttributedStringStyle` struct into a case in an enumeration, and allows the style object to be created by passing in an array of parts. This reads pretty nicely, and feels swifty, but it's a good amount of code to support.

```swift
let style = BonMot(
    .font(.fontB),
    .textColor(.colorB),
    .headIndent(10),
    .tailIndent(10)
)
```
<img width="785" alt="bonmot-part" src="https://cloud.githubusercontent.com/assets/114322/18294263/f2ffcb2a-7465-11e6-8993-cf8c51fe8926.png">

Which do you prefer, and why? Tell us your opinion below. Code samples are always appreciated, since they can make your point more clearly than any amount of prose.

**Comment by dostrander** [comment]
I think `BonMotC` is the most useful in general between the `BonMot` `BonMotI` and `BonMotC`. The MAIN reason being it's easy to pass this style object around to different methods or even have extensions on the style itself and just call titleAttributes(string) to have default title size, color, etc…

Also i'd say it's the most powerful in terms of leveraging the language and how extensible it is by being able to pass the style around rather then the BonChain or the attributes

(This was basically just copied from my other comment in FinePrint)

I do also like `BonMot` Part Construction especially when you don't have to do anything fancy because it looks the cleanest, however like you said it's a lot of code to support to I wouldn't necessarily say that it's the best foot forward

@ZevEisenberg @KingOfBrian

**Comment by KingOfBrian** [comment]
As we worked on the code, it became more obvious that the part style was the way to go. There is still a `derive` method on a style that will create a new style with the block format that should handle the behavior you liked @dostrander .
